src/toolkits/object_detection/object_detector.cpp

```diff
@@ -539,6 +554,132 @@ variant_map_type object_detector::perform_evaluation(gl_sframe data,
   return result_map;
 }
 
+std::unique_ptr<model_spec> object_detector::init_darknet_yolo() const {
+
+  std::unique_ptr<model_spec> nn_spec(new model_spec);
+
+  // Initialize a random number generator for weight initialization.
+  std::seed_seq seed_seq = { read_state<int>("random_seed") };
+  std::mt19937 random_engine(seed_seq);
+
+  // Initilize Layer 0
+  int num_features = 3;
+  xavier_weight_initializer conv0_init_fn(16*3*3, 16*3*3, &random_engine);
+  nn_spec->add_convolution(/* name */                "conv0_fwd",
+                           /* input */               "leakyrelu6_fwd",
+                           /* num_output_channels */ 16,
+                           /* num_kernel_channels */ num_features,
+                           /* kernel_height */       3,
+                           /* kernel_width */        3,
+                           /* stride_height */       1,
+                           /* stride_width */        1,
+                           /* padding */             padding_type::SAME,
+                           /* weight_init_fn */      conv0_init_fn);
+
+  // Append batchnorm7.
+  nn_spec->add_batchnorm(/* name */                  "batchnorm0_fwd",
+                         /* input */                 "conv0_fwd",
+                         /* num_channels */          16,
+                         /* epsilon */               0.00001f);
+
+  // Append leakyrelu7.
+  nn_spec->add_leakyrelu(/* name */                  "leakyrelu0_fwd",
+                         /* input */                 "batchnorm0_fwd",
+                         /* alpha */                 0.1f);
+
+  num_features = 16;
+  // Initilize Layer 1 to Layer 6
+  const std::map<int, int> idx_channels = {{1, 32}, {2,64}, {3, 128}, {4, 256}, {5,512}, {6, 1024}};
```

Comment:
Would an array make more sense here? 

Maybe something like:
`const int layer_idx_to_channels[] = {NULL, 32, 64, 128, 256, 512, 1024};`

Reply:
I'm mapping layer number to num of channels. I feel map makes it more readable. Let me know if you feel this isn't the right way to go about it.